source/common/router/router.cc

```diff
@@ -2026,15 +2026,18 @@ uint32_t Filter::numRequestsAwaitingHeaders() {
 
 bool Filter::checkDropOverload(Upstream::ThreadLocalCluster& cluster,
                                std::function<void(Http::ResponseHeaderMap&)>& modify_headers) {
+  if (!Runtime::runtimeFeatureEnabled("envoy.reloadable_features.enable_drop_overload")) {
```

Comment:
Please add this as part of the `if (cluster.dropOverload().value())` below.
This will couple the change with the code (so if later new code is added after the `if` scope, it will be used whether the runtime feature is enabled or not).

Reply:
I separate it just for a little bit easy to read.

I don't fully understand your comments on why it is better to combine the two if-conditions together.